data-raw/qc-markdowns/adult-holding-redd-and-carcass-surveys/feather-river/feather-carcass-survey-qc.Rmd

```diff
@@ -0,0 +1,539 @@
+  
+**Data Contact:** 
+  
+Any additional info?
+
+## Access Cloud Data
+
+```{r, eval=FALSE}
+# Run Sys.setenv() to specify GCS_AUTH_FILE and GCS_DEFAULT_BUCKET before running 
+# getwd() to see how to specify paths 
+# Open object from google cloud storage
+# Set your authentication using gcs_auth
+gcs_auth(json_file = Sys.getenv("GCS_AUTH_FILE"))
+# Set global bucket 
+gcs_global_bucket(bucket = Sys.getenv("GCS_DEFAULT_BUCKET"))
+
+# git data and save as xlsx
+```
+
+
+```{r}
+read_from_cloud <- function(carcass_data_type, year){
+  gcs_get_object(object_name = paste0("adult-holding-redd-and-carcass-surveys/feather-river/data/feather_carcass_",carcass_data_type, "_",year, ".csv"),
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = paste0("feather_carcass_",carcass_data_type,"_",year, ".csv"),
+               overwrite = TRUE)
+  data <- read.csv(paste0("feather_carcass_",carcass_data_type, "_", year, ".csv"))  
+}
+```
+
+### 2000 Chops and Tags
+*2000 and 2001 chops and tags are in different tables
+*2000 grilse has sex information
+
+```{r}
+feather_carcass_chops_2000 <- read_from_cloud("chops", 2000) %>% 
+  rename(adult_male = "male_chop",
+         adult_female = "female_chop",
+         sec = "sect") %>% 
+  pivot_longer(cols = c("adult_male", "adult_female", "grilse_male",
+                        "grilse_female"),
+               names_to = c("lifestage", "sex"),
+               names_sep ="_",
+               values_to = "chop_count") %>% 
+  mutate(chop_clip_status = "unknown",
+           tag_clip_status = "unknown")
+
+feather_carcass_tags_2000 <- read_from_cloud("tags", 2000) %>% 
+  mutate(tag_count = 1)
+
+# this only joins on "sex" - I don't think that is robust 
+# further, not sure that renaming recov_id is accurate here? that's why we are getting id values that are NA bc of the join
+# TODO: go back into 2000 .Rmd and figure out if there is tag data w/ ids and dates (better for joining on)
+feather_carcass_chops_and_tags_2000 <- full_join(feather_carcass_chops_2000, feather_carcass_tags_2000 ) %>%
+  select(-c(chop_env_id, egg_ret, fl, tag_num)) %>%
+  mutate(sex = case_when(
+    sex == "m" ~ "male",
+    sex == "f" ~ "female",
+    TRUE ~ sex
+  )) %>%
+  rename(id = "recov_id") %>% glimpse
+
+# does not contain tag_clip_status, chop_clip_status, or tag_count in data
+```
+
+### 2001 Chops and Tags
+```{r}
+feather_carcass_chops_2001 <- read_from_cloud("chops", 2001) %>% 
+  rename(adult_male = "male_chop",
+         adult_female = "female_chop",
+         grilse_unknown = "grilse",
+         sec = "sect") %>% 
+  pivot_longer(cols = c("adult_male", "adult_female", "grilse_unknown"),
+               names_to = c("lifestage", "sex"),
+               names_sep ="_",
+               values_to = "chop_count") %>% 
+  mutate(chop_clip_status = "unknown",
+           tag_clip_status = "unknown")
+
+feather_carcass_tags_2001 <- read_from_cloud("tags", 2001) %>% 
+  mutate(tag_count = 1)
+
```

Comment:
This comments applies to all if tag_count is NA then tag_clip_status should also be NA. If a variable isn't recorded I think we should use "not recorded" instead of "unknown". Unknown makes me think that it was observed but so much uncertainty that couldn't make distinction which is different than not collecting the variable.